internal/buildengine/engine.go

```diff
@@ -786,6 +835,58 @@ func (e *Engine) getDependentModuleNames(moduleName string) []string {
 	return maps.Keys(dependentModuleNames)
 }
 
+// processDeployments runs in a separate goroutine and processes deployments from the queue
+func (e *Engine) processDeployments(ctx context.Context) {
+	logger := log.FromContext(ctx)
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
```

Comment:
I preferred the other approach, as much as possible we should be trying to avoid using a ticker vs just reading directly from the channel. 100ms isn't much, but if they start poping up everywhere they eventually really start to slow things down.